### ISSUE TITLE
[BUG] Disable kcal from machinery issue

### ISSUE THREAD

**Issue by RickNY** [cagnulein/qdomyos-zwift#975]
**Describe the bug**
If the option "Disable kcal from machinery" is turned on, QZ utilizes the built-in formulas for calculating kcal.  However, if this option is turned OFF, it appears that QZ is ignoring the reported machinery kcal and still using its internal formulas.

**To Reproduce**
1. Go to Settings, Heart Rate Options - check that "Disable kcal from machinery" is turned OFF
2. Start treadmill.  Calories reported on tile do not match what is reported on treadmill.
3. Changing setting for "Disable kcal from machinery" to on reports the same calorie metrics in the tile

**Expected behavior**
If the treadmill is reporting kcal via BT (and in the debug log, it appears it is) and the option to "Disable kcal from machinery" is NOT enabled, QZ should utilize the reported kcal from the machinery.
  
**Additional context**

[debug-Sun_Oct_9_13_43_36_2022.log.txt](https://github.com/cagnulein/qdomyos-zwift/files/9742244/debug-Sun_Oct_9_13_43_36_2022.log.txt)

**Comment by cagnulein** [comment]
yes it will complicate a little because i have to parsing also the ftms chars, anyway it's possible. Thanks for the feedback
This issue is stricly about the horizon treadmills

**Comment by cagnulein** [comment]
thanks bot



Il giorno lun 24 ott 2022 alle ore 20:44 stale[bot] <
***@***.***> ha scritto:

**Comment by cagnulein** [comment]
yes bot

**Comment by cagnulein** [comment]
stupid bot

**Comment by cagnulein** [comment]
yes thanks for the reminder
